docs/react/guides/prefetching.md

```diff
@@ -36,30 +51,328 @@ const prefetchTodos = async () => {
+      Show Details
+    </button>
+  )
+}
+```
 
-[//]: # 'ExampleSetQueryData'
+## Prefetch in components
+
+Prefetching during the component lifecycle is useful when we know some child or descendant will need a particular piece of data, but we can't render that until some other query has finished loading. Let's borrow an example from the Request Waterfall guide to explain:
 
 ```tsx
-queryClient.setQueryData(['todos'], todos)
+function Article({ id }) {
+  const { data: articleData, isPending } = useQuery({
+    queryKey: ['article', id],
+    queryFn: getArticleById,
+  })
+
+  if (isPending) {
+    return 'Loading article...'
+  }
+
+  return (
+    <>
+      <ArticleHeader articleData={articleData} />
+      <ArticleBody articleData={articleData} />
+      <Comments id={id} />
+    </>
+  )
+}
+
+function Comments({ id }) {
+  const { data, isPending } = useQuery({
+    queryKey: ['article-comments', id],
+    queryFn: getArticleCommentsById,
+  })
+
+  ...
+}
+```
+
+This results in a request waterfall looking like this:
+
+```
+1. |> getArticleById()
+2.   |> getArticleCommentsById()
+```
+
+As mentioned in that guide, one way to flatten this waterfall and improve performance is to hoist the `getArticleCommentsById` query to the parent and pass down the result as a prop, but what if this is not feasible or desirable, for example when the components are unrelated and have multiple levels between them?
+
+In that case, we can instead prefetch the query in the parent. The simplest way to do this is to use a query but ignore the result:
+
+```tsx
+function Article({ id }) {
+  const { data: articleData, isPending } = useQuery({
+    queryKey: ['article', id],
+    queryFn: getArticleById,
+  })
+
+  // Prefetch
+  useQuery({
+    queryKey: ['article-comments', id],
+    queryFn: getArticleCommentsById,
+    // Optional optimization to avoid rerenders when this query changes:
+    notifyOnChangeProps: [],
+  })
+
+  if (isPending) {
+    return 'Loading article...'
+  }
+
+  return (
+    <>
+      <ArticleHeader articleData={articleData} />
+      <ArticleBody articleData={articleData} />
+      <Comments id={id} />
+    </>
+  )
+}
+
+function Comments({ id }) {
+  const { data, isPending } = useQuery({
+    queryKey: ['article-comments', id],
+    queryFn: getArticleCommentsById,
+  })
+
+  ...
+}
+```
+
+This starts fetching `'article-comments'` immediately and flattens the waterfall:
+
+```
+1. |> getArticleById()
+1. |> getArticleCommentsById()
+```
+
+You can also do this with suspense queries, but instead of using two `useSuspenseQuery` you'll have to use `useSuspenseQueries` to group the queries instead, just ignore the result from the query you are prefetching:
+
+```tsx
+const [articleResult] = useSuspenseQueries({
+  queries: [
+    { queryKey: ['article', id], queryFn: getArticleById },
+    // Prefetch
+    { queryKey: ['article-comments', id], queryFn: getArticleCommentsById },
+  ],
```

Comment:
a little gotcha here: Now, your component will only resume after suspending once both queries have finished. if `getArticleCommentsById` takes a long time or e.g. needs to go through retries, you'll see the suspense fallback for much longer than necessary. So I don't think this is the best idea after all :/

Reply:
Good point, and nice find! I'll rewrite this to use a `useQuery` for prefetching like the other one (above the suspense one), and explain why. Long term this is another good argument for a `usePrefetchQuery` or similar.